diem-move/diem-transactional-test-harness/src/diem_test_harness.rs

```diff
@@ -448,9 +503,81 @@ impl<'a> DiemTestAdapter<'a> {
+            parameters.sequence_number,
+            diem_transaction_builder::experimental_stdlib::encode_set_validator_operator_script_function(
+                validator_name.as_bytes().into(),
+                operator_account_addr,
+            ),
+            parameters.max_gas_amount,
+            parameters.gas_unit_price,
+            parameters.gas_currency_code,
+            parameters.expiration_timestamp_secs,
+            ChainId::test(),
+        )
+        .sign(
+            &validator_private_key,
+            validator_private_key.public_key().clone(),
+        )
+        .unwrap()
+        .into_inner();
+        self.run_transaction(Transaction::UserTransaction(txn))
+            .expect("Failed to set validator operator. This should not happen.");
+
+        // Step 4. Set validator config.
+        let parameters = self
+            .fetch_default_transaction_parameters(&operator_account_addr)
+            .unwrap();
+        let txn = RawTransaction::new(
+            operator_account_addr,
+            parameters.sequence_number,
+            diem_transaction_builder::experimental_stdlib::encode_register_validator_config_script_function(validator_account_addr, validator_private_key.public_key().to_bytes().to_vec(), vec![], vec![]),
+            parameters.max_gas_amount,
+            parameters.gas_unit_price,
+            parameters.gas_currency_code,
+            parameters.expiration_timestamp_secs,
+            ChainId::test(),
+        )
+        .sign(
+            &operator_private_key,
+            operator_private_key.public_key().clone(),
+        )
+        .unwrap()
+        .into_inner();
+        self.run_transaction(Transaction::UserTransaction(txn))
+            .expect("Failed to set validator config. This should not happen.");
 
-        self.run_transaction(txn)
-            .expect("Failed to create validator account. This should not happen.")
+        // Step 5. Add validator to validator set.
+        let parameters = self
+            .fetch_default_transaction_parameters(&diem_root_address())
+            .unwrap();
+        let txn = RawTransaction::new(
+                diem_root_address(),
+                parameters.sequence_number,
+                diem_transaction_builder::experimental_stdlib::encode_add_validator_and_reconfigure_script_function(
+                    0,
+                    validator_name.as_bytes().into(),
+                    validator_account_addr,
+                ),
+                parameters.max_gas_amount,
+                parameters.gas_unit_price,
+                parameters.gas_currency_code,
+                parameters.expiration_timestamp_secs,
+                ChainId::test(),
+            )
+            .sign(&GENESIS_KEYPAIR.0, GENESIS_KEYPAIR.1.clone())
+            .unwrap()
+            .into_inner();
+        self.run_transaction(Transaction::UserTransaction(txn))
+            .expect("Failed to add validator to validator set. This should not happen.");
```

Comment:
Yeah, it turned out the validator account creation procedure I previously implemented was barely sufficient. It's missing the subsequent steps (creating and setting the operator, registering the validator config and adding the validator to the validator set) so I had to fix it here.